prime-angular-frontend/src/app/modules/adjudication/shared/components/site-registration-container/site-registration-container.component.ts

```diff
@@ -81,6 +82,71 @@ export class SiteRegistrationContainerComponent implements OnInit {
       : this.deleteSite(record.siteId);
   }
 
+  public onApprove(siteId: number) {
+    const data: DialogOptions = {
+      title: 'Approve Site Registration',
+      message: 'Are you sure you want to approve this Registration?',
+      actionText: 'Approve Site Registration',
+      component: NoteComponent
+    };
+
+    let note: string;
```

Comment:
We could reduce the use of closures and keep the observable pure by piping `note` through the stream